app/controllers/course/assessment/question/programming_controller.rb

```diff
@@ -105,7 +119,24 @@ def update_question_setting
   end
 
   def destroy
-    if @programming_question.destroy
+    koditsu_question_id = @programming_question.acting_as.koditsu_question_id
+
+    success = @programming_question.class.transaction do
+      raise ActiveRecord::Rollback unless @programming_question.destroy
+
+      is_course_koditsu_enabled = current_course.component_enabled?(Course::KoditsuPlatformComponent)
+
+      if is_course_koditsu_enabled && @assessment.is_koditsu_enabled
+        @assessment.update!(is_synced_with_koditsu: false)
+        remove_question_from_assessment_in_koditsu(koditsu_question_id)
```

Comment:
Code is problematic. Coursemology cannot fail to delete if Koditsu fails.

Reply:
fixed.. the function `remove_question_from_assessment_in_koditsu` won't raise any error and instead return directly if the status returned by the API is not 200

Reply:
Any issue if Koditsu is not responding? i.e. no status code returned, instead, there was a timeout.